lib/secure_headers/headers/content_security_policy/source_expression.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+module SecureHeaders
+  class ContentSecurityPolicy
+    class SourceExpression
```

Comment:
I wonder if we should just call this a "expression" even something more general, rather than a "source expression".

My goal was to support just "source lists", as defined in the spec and implied in `content_security_policy.rb`. But for now we still need "source expression" parsing to allow report paths, as well as unquoted directives like `script-src`.